src/Sylius/Behat/Service/NotificationChecker.php

```diff
@@ -66,29 +67,47 @@ public function checkEditionNotification($resource)
     {
         $message = sprintf('%s has been successfully updated.', $this->humanizeResourceName($resource));
 
-        $this->checkSuccessNotificationMessage($message);
+        $this->checkNotification($message, NotificationType::success());
     }
 
     /**
-     * @param string $message
-     *
-     * @throws NotificationExpectationMismatchException
+     * {@inheritdoc}
      */
-    public function checkSuccessNotificationMessage($message)
+    public function checkNotification($message, NotificationType $type)
     {
-        if ($this->notificationAccessor->hasSuccessMessage() && $this->notificationAccessor->hasMessage($message)) {
+        if ($this->hasType($type) && $this->hasMessage($message)) {
             return;
         }
 
         throw new NotificationExpectationMismatchException(
-            'success',
+            $type,
             $message,
-            $this->notificationAccessor->getMessageType(),
+            $this->notificationAccessor->getType(),
             $this->notificationAccessor->getMessage()
         );
     }
 
     /**
+     * @param NotificationType $type
+     *
+     * @return bool
+     */
+    private function hasType(NotificationType $type)
+    {
+        return (string) $type === (string) $this->notificationAccessor->getType();
```

Comment:
That's quite a smell. Can `NotificationType` implement the flyweight pattern along with the existing static factory method, so every success notification type will share the same instance? Then we will be able to compare two types by `$type === $this->notificationAccessor->getType()`.